### ISSUE TITLE
'paths relative to ${prefix} in the pkg-config' hardly work (as understood)

### ISSUE THREAD

**Issue by stephan57160** [lu-zero/cargo-c#293]
Found PR #291 interesting, so ... tried to use it :-)

* Attempt 1
  Following current README:
    ```
    DESTDIR="${HOME}/mystuff"
    BUILD_PREFIX="/my/local"
    rm -rf "${DESTDIR}"
    cargo clean
    cargo cbuild \
	    --destdir="${DESTDIR}" \
	    --prefix="${BUILD_PREFIX}" \
	    --libdir="${BUILD_PREFIX}/lib"
    cargo ctest
    cargo cinstall \
	    --destdir="${DESTDIR}" \
	    --prefix="${BUILD_PREFIX}" \
	    --libdir="${BUILD_PREFIX}/subdir/lib"
    ```
    produces the following:
    ```
    prefix=/my/local
    exec_prefix=${prefix}
    libdir=/my/local/subdir/lib                       # NOT relative path
    includedir=${prefix}/include
    ```

* Attempt N
  As I am kind of playfull, I started to play with this and --libdir, --release and other combinations.
  It took a few attempts to find something looking good:
    ```
    DESTDIR="${HOME}/mystuff"
    BUILD_PREFIX="/my/local"
    rm -rf "${DESTDIR}"
    cargo clean
    cargo cbuild \
	    --destdir="${DESTDIR}" \
	    --prefix="${BUILD_PREFIX}"
    cargo ctest
    cargo cinstall \
	    --destdir="${DESTDIR}" \
	    --prefix="${BUILD_PREFIX}"
    ```
    finally produces something interesting:
    ```
    prefix=/my/local
    exec_prefix=${prefix}
    libdir=${exec_prefix}/lib                       # Relative path
    includedir=${prefix}/include
    ```
  But then, there is no possibility to have `${BUILD_PREFIX}/subdir/lib`

**Comment by stephan57160** [comment]
Below, a small script to help.
```
#!/bin/bash

set -e
set -x

# The following 4 vars should be fine to test a few combinations:
DESTDIR="${HOME}/mystuff"
BUILD_PREFIX="my/local"
LIBDIR="subdir/lib"
RELEASE="--release"


rm -rf "${DESTDIR}"
cargo clean
cargo cbuild \
	--destdir="${DESTDIR}" \
	--prefix="/${BUILD_PREFIX}" \
	--libdir="/${BUILD_PREFIX}/${LIBDIR}" \
	"${RELEASE}"
cargo ctest \
        "${RELEASE}"
cargo cinstall \
	--destdir="${DESTDIR}" \
	--prefix="/${BUILD_PREFIX}" \
	--libdir="/${BUILD_PREFIX}/${LIBDIR}" \
        "${RELEASE}"
cat "${DESTDIR}/${BUILD_PREFIX}/${LIBDIR}/pkgconfig/somelibrary.pc"
```

**Comment by lu-zero** [comment]
I'm not sure that's what @sdroege had in mind and I'm not sure what would be your use-case.

**Comment by sdroege** [comment]
We use it in GStreamer's cerbero build system. See https://gitlab.freedesktop.org/gstreamer/cerbero/-/merge_requests/1012 and https://gitlab.freedesktop.org/gstreamer/cerbero/-/merge_requests/1014.

We need relocatable pkg-config files there and as long as the libdir is inside the prefix this should work fine (and works fine for us). E.g. `--prefix dist --libdir dist/lib` (or with absolute paths) would work.

**Comment by stephan57160** [comment]
@sdroege : `as long as the libdir is inside the prefix this should work fine`, but not working for me.

With the below:
```
stephan@pc-stephan:cargo-c-testing$ cargo cinstall --destdir=${HOME}/mystuff --prefix=/usr/local/myapplication --libdir=/usr/local/myapplication/lib
```

I get this:
```
stephan@pc-stephan:cargo-c-testing$ more ~/mystuff/usr/local/myapplication/lib/pkgconfig/somelibrary.pc 
prefix=/usr/local/myapplication
exec_prefix=${prefix}
libdir=/usr/local/myapplication/lib
includedir=${prefix}/include
```

Note:
```
stephan@pc-stephan:~$ cargo install cargo-c
    Updating crates.io index
  Installing cargo-c v0.9.14+cargo-0.66          # Should be the correct version ?
   Compiling libc v0.2.137
   Compiling proc-macro2 v1.0.47
```

**Comment by sdroege** [comment]
Can you add some `println!` in the code added by https://github.com/lu-zero/cargo-c/pull/291 to see why it doesn't make it a relative path in your case?

**Comment by sdroege** [comment]
Yes that works fine for me.

```console
$ ../target/debug/cargo-cinstall cinstall --prefix dist --libdir dist/lib
[...]
$ cat dist/lib/pkgconfig/example_project.pc 
prefix=dist
exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include

Name: example_project
Description: 
Version: 0.1.0
Libs: -L${libdir} -lexample-project
Cflags: -I${includedir}/example-project-0.1
Libs.private:  -lgcc_s -lutil -lrt -lpthread -lm -ldl -lc

$ ../target/debug/cargo-cinstall cinstall --prefix (pwd)/dist --libdir (pwd)/dist/lib
[...]
$ cat dist/lib/pkgconfig/example_project.pc 
prefix=/home/slomo/Projects/rust/cargo-c/example-project/dist
exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include

Name: example_project
Description: 
Version: 0.1.0
Libs: -L${libdir} -lexample-project
Cflags: -I${includedir}/example-project-0.1
Libs.private:  -lgcc_s -lutil -lrt -lpthread -lm -ldl -lc

$ ../target/debug/cargo-cinstall cinstall --prefix (pwd)/dist --libdir (pwd)/dist/lib --destdir (pwd)/dest
[...]
$ cat dest/home/slomo/Projects/rust/cargo-c/example-project/dist/lib/pkgconfig/example_project.pc 
prefix=/home/slomo/Projects/rust/cargo-c/example-project/dist
exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include

Name: example_project
Description: 
Version: 0.1.0
Libs: -L${libdir} -lexample-project
Cflags: -I${includedir}/example-project-0.1
Libs.private:  -lgcc_s -lutil -lrt -lpthread -lm -ldl -lc
```

**Comment by sdroege** [comment]
> Installing cargo-c v0.9.14+cargo-0.66          # Should be the correct version ?

It's not in 0.9.14 but only git currently. The PR was merged 21 days ago, 0.9.14 was released 28 days ago.
